app/services/metrics/review_turnaround/per_user_project.rb

```diff
@@ -10,22 +11,42 @@ def call
       private
 
       def process
-        filtered_reviews_ids.each_slice(BATCH_SIZE) do |batch|
-          Events::Review.eager_load(:review_request).find(batch).lazy.each do |review|
+        filtered_reviews_ids.lazy.each_slice(BATCH_SIZE) do |batch|
+          Events::Review.eager_load(:review_request).find(batch).each do |review|
             entity = find_user_project(review.owner, review.pull_request.project)
             turnaround = calculate_turnaround(review)
 
-            create_metric(entity, turnaround)
+            create_or_update_metric(entity, turnaround)
           end
         end
+        calculate_avg
```

Comment:
I don't think this solves the problem we discussed. What if a user reviews several PRs for the same project in the same day? If that happens, you would call `create_or_update_metric(entity, turnaround)` several times for the same user-project pair (one for each review), and you would only keep the value that was processed last.

If you want we can hop on a call with @hvilloria and find a solution